### ISSUE TITLE
To_csv method is not working

### ISSUE THREAD

**Issue by bibs2091** [deepchem/deepchem#3060]
## 🐛 Bug

I was trying to save my dataset to a CSV file but deepchem is telling me that the the method to_csv doesn't exist for numpydataset or diskdataset. 

## To Reproduce
If you run the example in the official docs:
```
import numpy as np
X = np.random.rand(10, 10)
dataset = dc.data.DiskDataset.from_numpy(X)
dataset.to_csv('out.csv')  
```
You will have an error `"AttributeError: 'DiskDataset' object has no attribute 'to_csv'"`

## Environment

* OS: Macos
* Python version: 3.9
* DeepChem version: 2.6.1

**Comment by arunppsg** [comment]
`dataset` has to be NumpyDataset.
```
import numpy as np
X = np.random.rand(10, 10)
dataset = dc.data.NumpyDataset(X=X)
dataset.to_csv('out.csv')
```

If you want to use `to_csv` in DiskDataset, you should have deepchem nightly version installed - DiskDataset.to_csv was added recently.

**Comment by osbm** [comment]
In that case using `NumpyDataset.to_dataframe().to_csv` would help:
```python
import numpy as np
X = np.random.rand(10, 10)
dataset = dc.data.NumpyDataset(X=X)
dataset.to_dataframe().to_csv("out.csv")
```

**Comment by gusty1g** [comment]
Closing this since it is fixed in #3007 .
